crates/topos-metrics/src/server.rs

```diff
@@ -0,0 +1,41 @@
+use std::net::SocketAddr;
+
+use axum::{routing::get, Router, Server};
+use prometheus::{self, Encoder, TextEncoder};
+use tracing::info;
+
+#[derive(Default)]
+pub struct ServerBuilder {
+    serve_addr: Option<SocketAddr>,
+}
+
+impl ServerBuilder {
+    pub fn serve_addr(mut self, addr: Option<SocketAddr>) -> Self {
+        self.serve_addr = addr;
+
+        self
+    }
+
+    pub async fn build(
+        mut self,
+    ) -> Server<hyper::server::conn::AddrIncoming, axum::routing::IntoMakeService<Router>> {
+        let app = Router::new().route(
+            "/metrics",
+            get(|| async move {
+                let mut buffer = Vec::new();
+                let encoder = TextEncoder::new();
+
+                // Gather the metrics.
+                let metric_families = prometheus::gather();
+                // Encode them to send.
+                encoder.encode(&metric_families, &mut buffer).unwrap();
+
+                String::from_utf8(buffer.clone()).unwrap()
```

Comment:
Maybe we can just put this closure into a `pub` function of the `metrics` crate and define the server in `topos-tce-api` as it is the place where we have every api server definnition

Reply:
Hm, yeah... Then we split it in three different pieces: `topos-metrics`, `topos-tce-api`, and in `topos-tce`, we built and start the server. Does it then make sense to start the server also in `topos-tce-api`?

Reply:
`topos-metrics` is a little crate to handle metrics system, nothing more.
`topos-api` exposes external api structs.
`topos-tce` is the runtime of a TCE node.

For now, the `API` on port `3000` is just here to expose a `/metrics` which read the `topos-metrics` state. But at some point we could expose other things on that, like a `/status` or any other non-related metrics things. It's a matter of SOC.

Reply:
👍 I updated the PR!